src/version.h

```diff
@@ -19,6 +19,33 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 
 #pragma once
 
-extern const char *g_version_string;
-extern const char *g_version_hash;
-extern const char *g_build_info;
+#include "config.h"
+
+#if defined(__ANDROID__)
+	#include "android_version.h"
+	#include "android_version_githash.h"
+#elif defined(USE_CMAKE_CONFIG_H)
+	#include "cmake_config_githash.h"
+#endif
+
+#ifndef VERSION_GITHASH
+	#define VERSION_GITHASH VERSION_STRING
+#endif
+
+#define STRINGIFY(x) #x
+#define STR(x) STRINGIFY(x)
+
+constexpr const char* g_version_string = VERSION_STRING;
```

Comment:
wrong asterisk place and this must be static too

Reply:
also there is no benefit using constexpr here as it's a macro, there is no calculation it's replaced by compiler in the preprocessor